packages/ui/src/components/DataStory/clients/WorkspaceApiJSClient.tsx

```diff
@@ -74,12 +67,42 @@ const loadTrees = (key: string): Tree[] => {
   return parseDiagramTreeInfo(localStorage?.getItem(key) || '');
 }
 
+class ManualPromise {
```

Comment:
Interesting, what is the purpose of this class over `private appInitialized: boolean = false`?

Reply:
`appInitialized` is a Promise. Since setting `this.app` is asynchronous, calling `this.app!.descriptions()` will throw an error if `this.app` isn't set. Using `await this.appInitialized.promise;` allows for asynchronous waiting until `this.app` is fully set. If `appInitialized` were a boolean, the function below would not wait asynchronously `this.app` is set:

```typescript
getNodeDescriptions = async ({ path }): Promise<NodeDescription[]> => {
    await this.appInitialized.promise;
    const nodeDescriptions = this.app!.descriptions();
    // Other code
}
```

Reply:
Got it 👍